backend/cmm_builtins.ml

```diff
@@ -191,6 +191,235 @@ let bigstring_atomic_add size (arg1, arg2, arg3) dbg =
+  | "caml_float64x2_low_to_float" ->
+    let op = Cscalarcast (V128_to_scalar Float64x2) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_float32x4_low_of_float" ->
+    (* CR mslater: (SIMD) replace once we have unboxed float32 *)
+    let op = Cscalarcast (V128_of_scalar Float32x4) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_float32x4_low_to_float" ->
+    (* CR mslater: (SIMD) replace once we have unboxed float32 *)
+    let op = Cscalarcast (V128_to_scalar Float32x4) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int64x2_low_of_int64" ->
+    let op = Cscalarcast (V128_of_scalar Int64x2) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int64x2_low_to_int64" ->
+    let op = Cscalarcast (V128_to_scalar Int64x2) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int32x4_low_of_int32" ->
+    let op = Cscalarcast (V128_of_scalar Int32x4) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int32x4_low_to_int32" ->
+    let op = Cscalarcast (V128_to_scalar Int32x4) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int16x8_low_of_int" ->
+    (* CR mslater: (SIMD) replace once we have unboxed int16 *)
+    let op = Cscalarcast (V128_of_scalar Int16x8) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int16x8_low_to_int" ->
+    (* CR mslater: (SIMD) replace once we have unboxed int16 *)
+    let op = Cscalarcast (V128_to_scalar Int16x8) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int8x16_low_of_int" ->
+    (* CR mslater: (SIMD) replace once we have unboxed int8 *)
+    let op = Cscalarcast (V128_of_scalar Int8x16) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  | "caml_int8x16_low_to_int" ->
+    (* CR mslater: (SIMD) replace once we have unboxed int8 *)
+    let op = Cscalarcast (V128_to_scalar Int8x16) in
+    if_operation_supported op ~f:(fun () -> Cop (op, args, dbg))
+  (* Constants *)
+  | "caml_float32x4_const1" ->
+    (* CR mslater: (SIMD) replace once we have unboxed float32 *)
+    let f = const_float_args 1 args name |> List.hd in
+    let i = int64_of_float32 f in
+    let i = Int64.(logor (shift_left i 32) i) in
+    Some (Cconst_vec128 ({ low = i; high = i }, dbg))
+  | "caml_float32x4_const4" ->
+    (* CR mslater: (SIMD) replace once we have unboxed float32 *)
+    let i0, i1, i2, i3 =
+      match const_float_args 4 args name |> List.map int64_of_float32 with
+      | [i0; i1; i2; i3] -> i0, i1, i2, i3
+      | _ -> assert false
+    in
+    let low = Int64.(logor (shift_left i1 32) i0) in
+    let high = Int64.(logor (shift_left i3 32) i2) in
+    Some (Cconst_vec128 ({ low; high }, dbg))
+  | "caml_float64x2_const1" ->
+    let f = const_float_args 1 args name |> List.hd in
```

Comment:
(For consistency, we should probably
`assert` if the list has more than one
element.)

Reply:
Added a check to `const_*_args`